### PR TITLE
Support getter defs in CommonJS exports

### LINKED ISSUE #2510: Internal compiler when getters are used in module.exports when processing CommonJS modules
For example when consuming the [htmlparser2](https://github.com/fb55/htmlparser2) node module, the compiler crashes with:

```
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
  Node(SCRIPT): node_modules/htmlparser2/lib/index.js:1:0
var Parser = require("./Parser.js"),
  Parent: NULL
	at com.google.javascript.rhino.IR.mayBeExpression(IR.java:712)
	at com.google.javascript.rhino.IR.assign(IR.java:424)
	at com.google.javascript.jscomp.ProcessCommonJSModules$RewriteModule.expandObjectLitAssignment(ProcessCommonJSModules.java:885)
	at com.google.javascript.jscomp.ProcessCommonJSModules$RewriteModule.visitExport(ProcessCommonJSModules.java:792)
	at com.google.javascript.jscomp.ProcessCommonJSModules$RewriteModule.visit(ProcessCommonJSModules.java:678)
	at com.google.javascript.jscomp.NodeTraversal.handleScript(NodeTraversal.java:581)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:604)
	at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:305)
	at com.google.javascript.jscomp.NodeTraversal.traverseEs6(NodeTraversal.java:553)
	at com.google.javascript.jscomp.ProcessCommonJSModules.process(ProcessCommonJSModules.java:127)
	at com.google.javascript.jscomp.Compiler.processAMDAndCommonJSModules(Compiler.java:2107)
	at com.google.javascript.jscomp.Compiler.parseInputs(Compiler.java:1798)
	at com.google.javascript.jscomp.Compiler.parseForCompilationInternal(Compiler.java:1011)
	at com.google.javascript.jscomp.Compiler.access$300(Compiler.java:97)
	at com.google.javascript.jscomp.Compiler$7.call(Compiler.java:991)
	at com.google.javascript.jscomp.Compiler$7.call(Compiler.java:988)
	at com.google.javascript.jscomp.CompilerExecutor$2.call(CompilerExecutor.java:93)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	... 21 more
```

The offending part seem to be the `get` definitions in the module exports:
https://github.com/fb55/htmlparser2/blob/2e6c0e0f3f79cc268efc59e73b25d917c07d8493/lib/index.js#L15

### PR DESCRIPTION
**Pull request by kommen**
This fixes #2510

### CODE REVIEW

File: src/com/google/javascript/jscomp/ProcessCommonJSModules.java [hunk 1]
```diff
@@ -878,7 +878,7 @@ private void expandObjectLitAssignment(NodeTraversal t, Node export) {
           } else {
             value = IR.name(key.getString());
           }
-        } else if (key.isMemberFunctionDef()) {
+        } else if (key.isMemberFunctionDef() || key.isGetterDef()) {
```
Comment (ChadKillingsworth) [inline src/com/google/javascript/jscomp/ProcessCommonJSModules.java]: This blocks the crash, but doesn't produce correct output.  Either of the following would be acceptable transformations:

```js
/** @const */ var module$test = {
  get a() { return 1; }
};
```

or

```js
Object.defineProperties(module$test, {
  a: {
    get: function() { return 1; };
  }
});
```

**Comment by brad4d** [comment]
FYI, @kyshim is working on a simple 'report error instead of crash' fix for #2510 
We don't have time to do more than that right now.

@kommen we'll try out your fix after you address the problem @ChadKillingsworth pointed out.

**Comment by kommen** [comment]
All, thanks for the valuable and timely feedback. Much appreciated.

@ChadKillingsworth I've addressed your comments.
I'm still trying to make consuming `htmlparser2` (see #2510)  working. In my tests, the compiler now no longer crashes, but I get multiple warnings `JSC_COMMONJS_SUSPICIOUS_EXPORTS_ASSIGNMENT` and `JSC_JS_MODULE_LOAD_WARNING`

I'm not sure yet if this are different issues or if this PR still doesn't produce invalid output.

**Comment by brad4d** [comment]
@kommen 
I took a look at the `htmlparser2` link you had on #2510 
https://github.com/fb55/htmlparser2/blob/2e6c0e0f3f79cc268efc59e73b25d917c07d8493/lib/index.js#L15

I don't think we support using `require()` inside an exported object literal as is done there.
@ChadKillingsworth 
am I wrong about that?

If I'm not wrong, that's possibly a bigger problem that needs solving.

**Comment by ChadKillingsworth** [comment]
@brad4d I actually don't know. I'll run some test cases to see.

**Comment by kommen** [comment]
I rebased & squashed my commits onto current master, if you're still interested in merging this.

**Pull request closed without merging**